ResourceManager/GitPluginBase.cs

```diff
@@ -7,6 +7,16 @@
 
 namespace ResourceManager
 {
+    [UsedImplicitly]
+    public abstract class GitPluginWithoutSettingsBase : GitPluginBase
```

Comment:
Do we _really_ need the new type?
Is there a reason we can't pass a flag to the ctor indicating whether a plugin has settings or not?
```cs
    [Export(typeof(IGitPlugin))]
    public class GitImpactPlugin : GitPluginBase, IGitPluginForRepository
    {
        public GitImpactPlugin(bool hasSettings)
        {
            HasSettings = hasSettings;
        }

        public GitImpactPlugin()
            : this(true)
        {
        }

        public bool HasSettings { get; }
```